src/uwtools/rocoto.py

```diff
@@ -21,42 +19,43 @@
 
 
 def realize_rocoto_xml(
-    config_file: OptionalPath,
-    output_file: OptionalPath = None,
-) -> bool:
+    config: Union[OptionalPath, YAMLConfig], output_file: OptionalPath = None
+) -> str:
     """
-    Realize the Rocoto workflow defined in the given YAML as XML. Validate both the YAML input and
+    Realize the Rocoto workflow defined in the given YAML as XML, validating both the YAML input and
     XML output.
 
-    :param config_file: Path to YAML input file.
-    :param output_file: Path to write rendered XML file.
-    :return: Did the input and output files conform to theirr schemas?
+    :param config: Path to YAML input file (None => read stdin), or YAMLConfig object.
+    :param output_file: Path to write rendered XML file (None => write to stdout).
+    :return: An XML string.
     """
+    rxml = _RocotoXML(config)
+    xml = str(rxml)
+    assert validate_rocoto_xml_string(xml) is True
+    with writable(output_file) as f:
+        print(xml, file=f)
+    return xml
 
-    _, temp_file = tempfile.mkstemp(suffix=".xml")
 
-    _RocotoXML(config_file).dump(temp_file)
-
-    if not validate_rocoto_xml(input_xml=temp_file):
-        log.error("Rocoto validation errors identified in %s", temp_file)
-        return False
+def validate_rocoto_xml_file(xml_file: OptionalPath) -> bool:
+    """
+    Validate purported Rocoto XML file against its schema.
 
-    with open(temp_file, "r", encoding="utf-8") as f_in:
-        with writable(output_file) as f_out:
-            print(f_in.read(), file=f_out)
-    Path(temp_file).unlink()
-    return True
+    :param xml: Path to XML file (None => read stdin).
+    :return: Did the XML conform to the schema?
+    """
+    with readable(xml_file) as f:
+        return validate_rocoto_xml_string(xml=f.read())
 
 
-def validate_rocoto_xml(input_xml: OptionalPath) -> bool:
+def validate_rocoto_xml_string(xml: str) -> bool:
```

Comment:
This function, in combination with `validate_rocoto_xml_file()`, supports validating either on-disk XML files, e.g. via the CLI `uw rocoto validate` command; or in-memory XML strings, which we have via e.g. `uw rocoto realize` and want to validate before writing them to disk. Previously, the code was writing the string to a temp file, then validating that file.